Core/Src/main.c

```diff
@@ -65,6 +70,9 @@ void SystemClock_Config(void);
 int main(void)
 {
   /* USER CODE BEGIN 1 */
+  uint8_t rx_buf[3] = {0};
+  uint8_t tx_buf[3];
```

Comment:
Since this tx buffer never changes, you could initialize it to values when you first create it and skip the earlier step. eg. `uint8_t tx_buf[3] = {First_byte, second_byte, etc.};`